### ISSUE TITLE
God of War Ragnarök (2322010)

### ISSUE THREAD

**Issue by hjpaul7** [ValveSoftware/Proton#8107]
# Compatibility Report
- Name of the game with compatibility issues: God of War Ragnarok
- Steam AppID of the game: 2322010

## System Information
- GPU:  Nvidia 4090
- Video driver version: 560.35.03
- Kernel version: 6.10.10
- Link to full system information report: [System info gist](https://gist.github.com/hjpaul7/ea6405becfc2643394a0302ac8758d39)
- Proton version: `1721412568 experimental-9.0-20240719`

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

[steam-2322010.log](https://github.com/user-attachments/files/17062139/steam-2322010.log)

## Symptoms 
- Hit with "The program cannot start because the PlayStation PC SDK runtime is missing from you computer."

## Reproduction

- Launch game

**Comment by urbenlegend** [comment]
Same here:
![image](https://github.com/user-attachments/assets/e38e52ed-bd43-405e-9779-e0d9957d237c)

Sony ruining their PC efforts by mandating PSN. Is there a workaround for this?

**Comment by kisak-valve** [comment]
Hello @hjpaul7, it's fairly clear what the blocker is for this game. This is just a side note that the Proton log you attached is for a different game.

**Comment by hjpaul7** [comment]
> Hello @hjpaul7, it's fairly clear what the blocker is for this game. This is just a side note that the Proton log you attached is for a different game.

Yeah I realize, but Ghost of Tsushima included the PS SDK as well I believe and that game is able to run? Thanks, updated to correct Proton log.

**Comment by hjpaul7** [comment]
`SteamDeck=1 %command%` launch option allowed the game to launch. Compiling shaders currently..

**Comment by urbenlegend** [comment]
Seems like mouse look is deliberate disabled with SteamDeck=1? Anyone else seeing this? Controller seems to work okay after enabling Steam Input (typical Dualsense over BT issues).

**Comment by robotman40** [comment]
I suspect the PCSDK installer fails like in GoT? If so, can we work on fixing that?

**Comment by hjpaul7** [comment]
> Seems like mouse look is deliberate disabled with SteamDeck=1? Anyone else seeing this? Controller seems to work okay after enabling Steam Input (typical Dualsense over BT issues).

Also can't use mouse look. And I'm guessing with the `SteamDeck=1` option that disables something with Steam Input? My 8bitdo Ultimate controller will not swap the A/B buttons in game (use Nintendo layout toggled off in settings).

**Comment by urbenlegend** [comment]
Seems like there's a performance pitfall when Atmospherics is set higher than Medium and Krato's "flashlight" lights up in dark areas. Pretty reproducible in the dark sections early in the game. My Ryzen 3900x and Nvidia 3090 is barely maintaining 40fps in those areas.

**Comment by likarum** [comment]
`SteamDeck=1 %command%` launch option allowed the game to launch, but the screen remains black, but I'm able to access to the steam's overlay

**Comment by Dwarsen** [comment]
> `SteamDeck=1 %command%` launch option allowed the game to launch, but the screen remains black, but I'm able to access to the steam's overlay

Apparently this happens if you haven't linked Steam to your PSN account in a different Sony game prior to this one.

**Comment by likarum** [comment]
> > `SteamDeck=1 %command%` launch option allowed the game to launch, but the screen remains black, but I'm able to access to the steam's overlay
> 
> Apparently this happens if you haven't linked Steam to your PSN account in a different Sony game prior to this one.
Sadly sony is not agree to let me link my steam account with them...
error	"invalid_request"
error_code	4099
error_description	"Parameter 'client_id' is malformed"

**Comment by Dwarsen** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/8107#issuecomment-2362315939

How are you trying to do it? 

For me, the only thing that ended up working was to use Returnal, ask that game to open a web browser with a link (A QR code with the PS App would fail), and then send that link to my phone and open it in a mobile browser and log in. Then I could link it. The browser link wouldn't work on my desktop regardless of which browser I used, and I also could not login by going directly to the Playstation Store. I got a different error from you though.

**Comment by likarum** [comment]
My gosh... 
When I try to connect to sony network the url is: https://www.playstation.com/fr-fr/pc-registration/?auth_url..... with a defined client_id and a redirect_uri, but when I clicked on continue.... their value are "null"
I replaced null by the value of the first url.... and "Voila"

And I agree with you... now I'm able to access to the game...

**Comment by robotman40** [comment]
Just curious, but does the PlayStation Overlay work?

**Comment by Dwarsen** [comment]
> Just curious, but does the PlayStation Overlay work?

Nope.

**Comment by robotman40** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/8107#issuecomment-2362337471

What happens when you try to open it?

**Comment by Dwarsen** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/8107#issuecomment-2362337750

It doesn't initialize when you pass SteamDeck=1 to the game. If you don't pass that variable, the game won't start.

Just to add, I don't see anything at all about the overlay anywhere in the game at all. No popups, no button-combo, no nothing.

**Comment by robotman40** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/8107#issuecomment-2362338901

We'd have to try getting the pspcsdk to install, since it apparently fails due to a crash in kernel32.dll in GoT's case.

But even then, it may end up blocking Wine as it did with GoT and Concord.

**Comment by Dwarsen** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/8107#issuecomment-2362344934

Yeah, the PS SDK is the issue here, but I'm at least glad the developers allowed the integration to be entirely bypassed, I was worried they wouldn't. It's just unfortunate they still require your account to be linked, even though there is no way to do that in this game on Linux (as far as I can tell anyway).

In fact, looking at a steam deck reddit thread the game even has an automatic preset for Steam Deck, so clearly the devs wanted it to be playable on that device at least.

**Comment by sevansup** [comment]
Hello. I wanted to report that while the game now "works" on Steam Deck out of the box, at least in the USA region with a previously linked PSN/Steam account, the game doesn't work in Offline Mode, nor with airplane mode enabled/wifi disabled. It crashes with an error report pop-up that allows you to send an error report to the game's developers.

Is offline mode being prohibited, or is this a bug?

**Comment by matte-schwartz** [comment]
Hmm, is anyone else seeing their HDR settings reset every time you launch the game? Trying to figure out if it's a game issue or a proton issue, but every single launch I need to go into Settings -> Graphics & Camera -> Screen Calibration and manually enable HDR again.

**Comment by sevansup** [comment]
Heads up that offline mode now seems to work, sort of, if you can hold steam + the trackpads to change the input into mouse mode and skip past the PSN login message that appears in offline mode by accepting it. However, once you get to the main menu no controls are recognized. Kind of opposite to how SteamDeck=1 was disabling mouse input. That seems to be the next hurdle: something related to the game's handling of Steam Input based upon whether or not you're using a Deck.

**Comment by satarasov** [comment]
Is there any workaround to fix mouse lookup in game with `SteamDeck=1` parameter ?
Without using a controller :)

**Comment by Gumsax** [comment]
guys I have the same problem in my steam deck, try the steamdeck=1 and it doesn't work, try linking my steam account to playstation from another game and I still get the same mistake on the page, I'm from Argentina and I thought it was a problem of the region but I see that no, some still have the same problem, you've solved it in a different way?

**Comment by alosarjos** [comment]
I'm curious on Valve position here. Are games allowed to "run" on SteamDeck only and not regular PCs with Linux by default?

**Comment by ClaudeLib** [comment]
Ah, so that's why the game stays on black screen on startup and upon exiting I see a browser window with PSN login page open. The game indeed expects me to log in and link the accounts. And if I use SteamDeck=0 (I play on ChimeraOS, which has it set to 1, I believe), it just gives me the "PS PC SDK missing" error.

**Comment by urbenlegend** [comment]
> I'm curious on Valve position here. Are games allowed to "run" on SteamDeck only and not regular PCs with Linux by default?

I am sure Valve is okay with regular Linux PCs, especially since the difference between SteamOS's Arch-base and regular Linux is minimal at most. It's Sony here who's doing the arbitrary differentiation by relying on `SteamDeck=1` to toggle off the PSN SDK.

**Comment by alosarjos** [comment]
> > I'm curious on Valve position here. Are games allowed to "run" on SteamDeck only and not regular PCs with Linux by default?
> 
> I am sure Valve is okay with regular Linux PCs, especially since the difference between SteamOS's Arch-base and regular Linux is minimal at most. It's Sony here who's doing the arbitrary differentiation by relying on `SteamDeck=1` to toggle off the PSN SDK.

I meant, not sure the position on Valve allowing publishers to do this differentiation, since the point of Valve is also allowing 3rd party machines to have SteamOS without being Steam Decks

**Comment by mhmarf** [comment]
Sorry if it sounds moronic, but can we do something equivalent to workaround below on Linux to make mouse work?
[mouse fix on windows](https://steamcommunity.com/app/2322010/tradingforum/4843147950365271552/)

**Comment by alosarjos** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/8107#issuecomment-2364638634

I remember this from back the day

https://www.gamesindustry.biz/valve-says-it-will-not-allow-exclusive-titles-for-steam-deck

**Comment by mhmarf** [comment]
Many on Windows also cannot move camera using mouse, their fix according to this [steam discussions thread](https://steamcommunity.com/app/2322010/discussions/0/4843147950364370292/) is to install [Microsoft.GameInput](https://www.nuget.org/packages/Microsoft.GameInput). Is this possible for Linux?

> GoWR uses Microsoft GameInput for handling camera mouse input and it fails to initialize on some machines.

**Comment by dominikzogg** [comment]
@mhmarf i tried to run the .msi but it does not install using the same WINEPREFIX and proton version. It cannot be installed.

**Comment by thesacredsin** [comment]
> @mhmarf i tried to run the .msi but it does not install using the same WINEPREFIX and proton version. It cannot be installed.

I installed it, but it does not fix it for me, there is also a mention of a service that needs to be set to auto, but it is not present on Linux(Wine/Proton)

**Comment by mhmarf** [comment]
@thesacredsin How did you install it? I could not install it. I first used dotnet45, then wine-mono 9.2, finally dotnet48 (I'm not sure if these are needed...), then went to `wine uninstaller` -> install -> and chose this .msi file but kept getting this error (Wine 9.16-staging Nobara 40):
```
0140:fixme:file:NtLockFile I/O completion on lock not implemented yet
0140:fixme:ntdll:NtQuerySystemInformation info_class SYSTEM_PERFORMANCE_INFORMATION
014c:fixme:reg:NtQueryLicenseValue License key L"Kernel-ProductInfo" not found
00c4:err:service:process_send_start_message service L"GameInput Service" failed to start
0140:err:msi:ITERATE_StartService failed to start service L"GameInput Service" (1053)
0140:err:msi:execute_script Execution of script 0 halted; action L"StartServices" returned 1627
0140:err:msi:ITERATE_Actions Execution halted, action L"InstallFinalize" returned 1627
0140:fixme:msi:internal_ui_handler internal UI not implemented for message 0x0b000000 (UI level = 5)
0140:fixme:msi:internal_ui_handler internal UI not implemented for message 0x0b000000 (UI level = 5)
0140:err:msi:ITERATE_Actions Execution halted, action L"ExecuteAction" returned 1627
```
Though after trying many times, somehow when I used wine uninstaller, I indeed saw GameInput in there lol. I don't know how it got in there, but I went ahead and tested the game and mouse was still not fixed... I then removed it and no longer can "accidentally" install it... I'm not sure if it was correctly installed because I keep seeing these errors.

**Comment by thesacredsin** [comment]
I compiled latest Proton Tricks and then used Run arbitrary executable

**Comment by mhmarf** [comment]
Thank you! it installed easily. Yet mouse still not working...

Edit: I think GameInput is not working at all even though it gets installed... I tried to "modify" it (protontricks -> run uninstaller -> modify) and saw these errors:
```
016c:err:service:process_send_start_message service L"GameInput Service" failed to start
0164:err:msi:ITERATE_StartService failed to start service L"GameInput Service" (1053)
0164:err:msi:execute_script Execution of script 0 halted; action L"StartServices" returned 1627
0164:err:msi:ITERATE_Actions Execution halted, action L"InstallFinalize" returned 1627
```

**Comment by mhmarf** [comment]
> there is also a mention of a service that needs to be set to auto, but it is not present on Linux(Wine/Proton)

Can this be done in protontricks -> regedit? I found **GameInput Service** in there (HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services) :
![reged](https://github.com/user-attachments/assets/6d345378-9da9-432e-94e5-d91fbc418758)

**Comment by thesacredsin** [comment]
2 means auto I believe so it is already as it should be. Strange as I looked there too previously and did not find it https://www.dedoimedo.com/computers/wine-services.html

**Comment by mhmarf** [comment]
I just checked PROTON_LOG and indeed GameInput Service is being loaded (on regedit it is set to "2"):
```
17960.410:0054:0058:trace:loaddll:build_module Loaded L"C:\\Program Files (x86)\\Microsoft GameInput\\x64\\gameinputsvc.exe" at 0000000140000000: native
```

but is not starting:
```
17960.570:0030:0034:err:service:process_send_start_message service L"GameInput Service" failed to start
17960.570:0030:0034:fixme:service:scmdatabase_autostart_services Auto-start service L"GameInput Service" failed to start: 1053
```
If set to "0" (which is "boot") on regedit:
```
18396.029:0030:0034:err:service:validate_service_config Service L"GameInput Service" - SERVICE_BOOT_START and SERVICE_SYSTEM_START are only allowed for driver services
18396.029:0030:0034:err:service:scmdatabase_load_services Invalid configuration of service L"GameInput Service" - skipping
```
If set to "1" which is "system" (same message as "0"):
```
18654.803:0030:0034:err:service:validate_service_config Service L"GameInput Service" - SERVICE_BOOT_START and SERVICE_SYSTEM_START are only allowed for driver services
18654.803:0030:0034:err:service:scmdatabase_load_services Invalid configuration of service L"GameInput Service" - skipping
```
Finally, if set to "3" (which is "manual"), it won't even get loaded. it does not exist in PROTON_LOG.

**Comment by Hadrianneue** [comment]
FSR3 Frame Generation is very crash prone in this game, especially when using fullscreen mode, X11 or wayland, it doesn't seem to affect gamescope as much however, it usually crashes or freezes spamming these log lines:
```

46640.547:0128:0238:warn:threadname:NtSetInformationThread Thread renamed to L"DxUploadService"
46640.547:0128:023c:warn:threadname:NtSetInformationThread Thread renamed to L"vkd3d_fence"
46640.547:0128:0240:warn:threadname:NtSetInformationThread Thread renamed to L"vkd3d_queue"
46640.563:0128:024c:warn:seh:dispatch_exception EXCEPTION_WINE_NAME_THREAD exception (code=406d1388) raised
46640.961:0128:012c:warn:vkd3d-proton:vkd3d_get_image_allocation_info: Padding allocation requirements. Requested alignment 4096 < 65536 (dim 4, 128 x 128 x 128, 1 levels, 1 samples, fmt #27, flags #4).
46640.967:0128:012c:warn:vkd3d-proton:d3d12_resource_validate_texture_alignment: Invalid resource alignment 0x1000 (required 0x10000).
46641.359:0128:01c4:warn:vkd3d-proton:d3d12_pipeline_state_init_graphics_create_info: DSV format is DXGI_FORMAT_UNKNOWN.
46641.363:0128:012c:err:msvcrt:_wassert (L"!status && \"vkCreateWin32SurfaceKHR\"",L"../src-wine/dlls/winevulkan/loader_thunks.c",3252)
46641.364:0128:01b4:warn:vkd3d-proton:d3d12_pipeline_state_init_graphics_create_info: Ignoring stencilTestEnable due to lack of stencil aspect.
46641.377:0128:026c:err:ole:CoUninitialize Mismatched CoUninitialize
```

btw i tried with no_staggered_submit, crashes all the same.

**Comment by ilovethensa** [comment]
Any updates on this?

**Comment by matte-schwartz** [comment]
@Hadrianneue what Mesa version?

**Comment by Hadrianneue** [comment]
> @Hadrianneue what Mesa version?

mesa-git

**Comment by matte-schwartz** [comment]
I get a somewhat similar crash without frame-gen, after around ~40 minutes on mesa-git. Does not seem to happen for me at all on latest mesa stable (at least not yet...). No mention of `err:msvcrt:_wassert (L"!status && \"vkCreateWin32SurfaceKHR\"",L"../src-wine/dlls/winevulkan/loader_thunks.c",3252)` in my own proton log though: [steam-2322010.tar.gz](https://github.com/user-attachments/files/17108562/steam-2322010.tar.gz)

One time it crashed both gamescope and Xwayland, the second time (which is what my Proton log is of) only the game crashed and it gave me the game's own error reporter pop-up. 

```
  Host: threadripper-arch Kernel: 6.11.0-6-cachyos-lto arch: x86_64 bits: 64
  Desktop: KDE Plasma v: 6.1.5 Distro: CachyOS
Graphics:
  Device-1: Advanced Micro Devices [AMD/ATI] Navi 31 [Radeon RX 7900 XT/7900
    XTX/7900M] driver: amdgpu v: kernel
  Display: wayland server: X.org v: 1.21.1.13 with: Xwayland v: 24.1.2
    compositor: kwin_wayland driver: X: loaded: modesetting dri: radeonsi
    gpu: amdgpu resolution: 5120x1440
  API: EGL v: 1.5 drivers: kms_swrast,radeonsi,swrast
    platforms: gbm,wayland,x11,surfaceless,device
  API: OpenGL v: 4.6 compat-v: 4.5 vendor: amd mesa v: 24.3.0-devel
    renderer: AMD Radeon RX 7900 XTX (radeonsi navi31 LLVM 18.1.8 DRM 3.58
    6.11.0-6-cachyos-lto)
  API: Vulkan v: 1.3.295 drivers: radv,llvmpipe surfaces: xcb,xlib,wayland
```

**Comment by dougg0k** [comment]
https://github.com/AntiMicroX/antimicrox

![Screenshot from 2024-09-24 12-44-49](https://github.com/user-attachments/assets/a1daf5b1-3e1d-46c6-b051-036b81b46bf5)

**Comment by sevansup** [comment]
Is anyone investigating the issue where no controller input is recognized in offline mode on Steam Deck? Built in Deck controls, SteamOS.

**Comment by nipnipj** [comment]
I cannot move the camera with my mouse :(

**Comment by NiahSSBM** [comment]
Someone made a mod to remove the Playstation SDK runtime requirement, but it's since been taken down.
http://web.archive.org/web/20240924200559/https://github.com/iArtorias/nopssdk

I managed to save a copy before it was taken down, but I don't know if I'm allowed to distribute it.
I was hoping this would allow the game to start without the `SteamDeck=1`  workaround, and maybe the mouse would work, but the game crashes on startup and I get a crash report screen.

**Comment by ahmed4119** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/8107#issuecomment-2372583748

I tried adding the flags they mention to the configs.main.ini but it didn't reproduce your problem, for me it just asks for the PlayStation SDK is not found.
I need to mention that I am using a cracked version of the game and those DLLs I downloaded are new and do not replace anything that is not already in the directory and the versions of the DLLs I have are actually older (1.0.2).

**Comment by ahmed4119** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/8107#issuecomment-2372583748

If anyone has had any luck with NoPSSDK, please tell us how.

**Comment by NiahSSBM** [comment]
My mod version is also 1.0.2

My game was purchased through steam and the mod files are also net new.
My startup args are `SteamDeck=0 WINEDLLOVERRIDES="version=n,b" %command%`

![image](https://github.com/user-attachments/assets/3142dc9f-3a69-4eaa-b7c1-b2d980c9ec5e)

Also sorry if this isn't the place to be discussing mods the developer clearly does not want us playing with.

**Comment by mhmarf** [comment]
When they surrounded him, taking both his nexusmods and github page down, in his last final stand he released a final 1.0.3 version that actually works flawlessly with Linux, that is, the game launches without `SteamDeck=1` with no problem. But the mouse problem STILL remains... so SteamDeck=1 has nothing to do with this mouse problem and it is basically PSN=1 (lol)... This is a game problem and related to the dreadful GameInput.

**Comment by mhmarf** [comment]
Update on GameInput:
I've noticed one error in PROTON_LOG that accompanies GameInput (and never exists without it):
`5646.116:0054:0058:fixme:reg:NtQueryLicenseValue License key L"Kernel-ProductInfo" not found`

GameInput needs this "Kernel-ProductInfo"... whatever it is... it immediately fails to start after that:
```
5646.116:0054:0058:fixme:reg:NtQueryLicenseValue License key L"Kernel-ProductInfo" not found
5646.116:0030:0034:err:service:process_send_start_message service L"GameInput Service" failed to start
5646.116:0030:0034:fixme:service:scmdatabase_autostart_services Auto-start service L"GameInput Service" failed to start: 1053
```

**Comment by sevansup** [comment]
The mouse problem above is also related to the offline issue with Steam Deck where it doesn't recognize input, but will recognize controls if you bind them to mouse and keyboard buttons--but also no mouse input. Again, only in offline mode. There is something wonky with Steam Input on this game. Actually it's an issue on Windows too if you try to put the game in offline mode, it won't recognize your controller on Windows unless you disable Steam Input.

**Comment by alasky17** [comment]
> I get a somewhat similar crash without frame-gen, after around ~40 minutes on mesa-git. Does not seem to happen for me at all on latest mesa stable (at least not yet...). No mention of `err:msvcrt:_wassert (L"!status && \"vkCreateWin32SurfaceKHR\"",L"../src-wine/dlls/winevulkan/loader_thunks.c",3252)` in my own proton log though: [steam-2322010.tar.gz](https://github.com/user-attachments/files/17108562/steam-2322010.tar.gz)
> 
> One time it crashed both gamescope and Xwayland, the second time (which is what my Proton log is of) only the game crashed and it gave me the game's own error reporter pop-up.
> 
> ```
>   Host: threadripper-arch Kernel: 6.11.0-6-cachyos-lto arch: x86_64 bits: 64
>   Desktop: KDE Plasma v: 6.1.5 Distro: CachyOS
> Graphics:
>   Device-1: Advanced Micro Devices [AMD/ATI] Navi 31 [Radeon RX 7900 XT/7900
>     XTX/7900M] driver: amdgpu v: kernel
>   Display: wayland server: X.org v: 1.21.1.13 with: Xwayland v: 24.1.2
>     compositor: kwin_wayland driver: X: loaded: modesetting dri: radeonsi
>     gpu: amdgpu resolution: 5120x1440
>   API: EGL v: 1.5 drivers: kms_swrast,radeonsi,swrast
>     platforms: gbm,wayland,x11,surfaceless,device
>   API: OpenGL v: 4.6 compat-v: 4.5 vendor: amd mesa v: 24.3.0-devel
>     renderer: AMD Radeon RX 7900 XTX (radeonsi navi31 LLVM 18.1.8 DRM 3.58
>     6.11.0-6-cachyos-lto)
>   API: Vulkan v: 1.3.295 drivers: radv,llvmpipe surfaces: xcb,xlib,wayland
> ```

Could you retest experimental bleeding-edge and confirm if this is fixed?  (you will need to select the bleeding-edge-9.0 beta branch for Proton Expeirmental)

**Comment by robotman40** [comment]
I'm curious, but can anyone try copying the PSPCSDK files from a Windows install to this game's Proton prefix and see if it gets it to launch without the SteamDeck environment variable?

**Comment by mhmarf** [comment]
Patch 3 released!!
![mouse](https://github.com/user-attachments/assets/041a4f50-0a60-4cb3-b3f5-bf73a4c4d85d)
![5932ab955096a0cc2f180f3c9a1a25d0](https://github.com/user-attachments/assets/d45879ce-e932-4043-9af7-04cd16fd801b)

**Comment by urbenlegend** [comment]
Can confirm mouse controls now work on my Arch box after patch 3

**Comment by mozo78** [comment]
Yep, everything is working fine now.

**Comment by sevansup** [comment]
I refunded the game due to input not working in offline mode on Steam Deck. Could someone test and see if this fixes that too? Seems somewhat related.

**Comment by matte-schwartz** [comment]
@alasky17 I'm away from my problematic rig for a couple of days but I'll get back to you when I'm home to check regarding the game freezes/crashes. 

@sevansup I'm unable to even launch the game while I'm in offline mode while on the latest patch with my Steam Deck, but both the game and the controls do work if I start online and then go into airplane mode.

**Comment by matte-schwartz** [comment]
@alasky17 I have not had any crashes on mesa-git with proton experimental bleeding-edge after about ~8 hours of gameplay.

**Comment by carloscheddar** [comment]
Has anybody been able to get DualSense haptics working on this game? I'm able to get rumble working via Steam Input but then I lose the adaptive trigger functionality.

**Comment by robotman40** [comment]
> I'm curious, but can anyone try copying the PSPCSDK files from a Windows install to this game's Proton prefix and see if it gets it to launch without the SteamDeck environment variable?

Pinging this again, but would fixing the PSPCSDK be worth the effort? Like I know it's bypassable, but I think having an out-of-the-box experience with all features would be nice.

And according to a report on ProtonDB, someone did copy the PSPCSDK from a Windows PC to fix the issue of the game not working offline and according to them, it works. So... It might not be blocked anymore.

**Comment by mozo78** [comment]
It doesn't work. For PC the only way is the cracked version for now.

**Comment by robotman40** [comment]
> It doesn't work. For PC the only way is the cracked version for now.

Does it give the "Wine is not supported" message that it gave with Ghost of Tsushima?

Also, I don't want to play moderator, but I would advise not mentioning cracked games.

**Comment by mozo78** [comment]
Both Ghost of Tsushima and GoWR are working fine.

**Comment by robotman40** [comment]
> Both Ghost of Tsushima and GoWR are working fine.

No, I mean does copying the pspcsdk from a Windows install get it to run without `SteamDeck=1` on desktop Linux?

**Comment by mozo78** [comment]
With the cracked version there's no need to do anything.

**Comment by robotman40** [comment]
> With the cracked version there's no need to do anything.

Again, you're not answering my question

**Comment by mozo78** [comment]
Already answerd two times.  With the cracked version there's no pspcsdk check.

**Comment by ClaudeLib** [comment]
I don't think cracked versions can be discussed here. Not to mention they probably use the third party NoPSSDK bypass, which may or may not break with the next game update.

**Comment by Saladin1812** [comment]
For me it crashes after a few seconds infact it crashes my whole desktop. 

tested with flatpak on Vanilla OS
also tested with nixOS with system package

Doesn't seem to be a mesa specific or distro specific issue.. Also how to get logs for this.. as ` PROTON_LOG=1` generated file is empty because it crashes everything. 

GPU - 5700 xt
CPU Ryzen 5 3600x

**Comment by galacticaledge** [comment]
https://support.sms.playstation.com/hc/en-us/articles/30412721909261-PlayStation-SDK-Runtime-Missing-Error

Can anyone try to manually install the PlayStation PC SDK Runtime in GOW:R's prefix? I don't own the game myself, so I can't try.

**Comment by SniperJoe** [comment]
For those who are still curious, just copying PS SDK files will not change anything. I guess the .msi installer does some other magic, like registry keys setting, etc.

Installing the .msi crashes because the msiexec executes "CustomAction" as it reffered in MSI Installer API. This "CustomAction" is a call to "Binary.RuntimeInstallerCustomAction" dll in the .msi, which calls "runtime.dll" in the same .msi (and SDK). But "runtime.dll" ends up executing this:
```C
MEMORY[0x200000000](a1, 3764846851i64, a1, a2) //in C
```
```asm
call    near ptr 200000000h ; in asm
```
I dunno why it's even placed there, that's a question to Sony developers. But it obviously segfaults. Possibly, execution should NEVER reach it, and if it does, it means that something went terribly wrong. Maybe some conditions within Wine are different from Windows and so the execution actually reaches this instruction. But that's all I could figure out after a day of debugging.

**Comment by SniperJoe** [comment]
Ok, I have some good news. I was able to modify the .msi and remove the halting "CustomAction" from it. 

https://drive.google.com/file/d/1zkUx_UCsp60PO972gRZLHq185bRdu3ls/view?usp=sharing

Replace your PsPcSdkRuntimeInstaller.msi with this and install it to your GoW prefix. After that you shall be able to run the game.

**Comment by Hadrianneue** [comment]
> Ok, I have some good news. I was able to modify the .msi and remove the halting "CustomAction" from it.
> 
> https://drive.google.com/file/d/1zkUx_UCsp60PO972gRZLHq185bRdu3ls/view?usp=sharing
> 
> Replace your PsPcSdkRuntimeInstaller.msi with this and install it to your GoW prefix. After that you shall be able to run the game.

interesting, so that allows installing and running the ps overlay or just makes it skip pssdk check?

**Comment by SniperJoe** [comment]
> > Ok, I have some good news. I was able to modify the .msi and remove the halting "CustomAction" from it.
> > https://drive.google.com/file/d/1zkUx_UCsp60PO972gRZLHq185bRdu3ls/view?usp=sharing
> > Replace your PsPcSdkRuntimeInstaller.msi with this and install it to your GoW prefix. After that you shall be able to run the game.
> 
> interesting, so that allows installing and running the ps overlay or just makes it skip pssdk check?

pssdk check is definitely not skipped because the game did not start until I installed the .msi. About PS overlay - on my machine it suggests to login into PSN account, but when I pressed "yes", it instantly thanked me for joining and the thanks popup showed up simultaniously with game startup configuration popup. So I had two "press enter to confirm" on my screen at once lol. Need someone else to test and see if it has the same behavior on other machines.

P.S. Do not run the game with DualShock 4 connected (probably DualSense too), the game instantly crashes in such case.

**Comment by SniperJoe** [comment]
For those who are interested in running the game with Sony gamepad,  after 4 days of debugging & disassembling I came to a solution:
1. Open regedit for the prefix you run the game on
2. Go to HKEY_LOCAL_MACHINE\System\CurrentControlSet\Enum\HID 
3. In folder under name like "VID_054C&PID_{your_device_pid}", open the first child folder (it's name must be like "256&00:00:00:00:00:00&0&0&0") 
4. Create a string parameter with name "ContainerId" and value "{e27ebefb-ddd4-11ef-abea-40b076a40d44}" in the folder

**Comment by SniperJoe** [comment]
So for now, the game runs and can be correctly controlled with either mouse, xinput or ds4 gamepad. 
But it has a lot of stuttering during run. Once in 2 or 3 seconds it freezes for about a second. More than that, while the game is freezed for that second, the whole system is freezed too. 
I've modified Wine source code to log dll calls that take over 100ms, but could not catch any. 
@kisak-valve can you please give me some ideas how to debug/profile it? 🙏🏻

**Comment by galacticaledge** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/8107#issuecomment-2632359448

So you're saying it gets the runtime working, but the overlay itself doesn't work properly?

Also, does this workaround work in other games like GoT?

**Comment by Saladin1812** [comment]
> So for now, the game runs and can be correctly controlled with either mouse, xinput or ds4 gamepad. But it has a lot of stuttering during run. Once in 2 or 3 seconds it freezes for about a second. More than that, while the game is freezed for that second, the whole system is freezed too. I've modified Wine source code to log dll calls that take over 100ms, but could not catch any. [@kisak-valve](https://github.com/kisak-valve) can you please give me some ideas how to debug/profile it? 🙏🏻

Experiencing the same issue.. and then my whole system crashes.. not sure if my issue is the exact same.. I just can't get into the game.. tried different kernel, mesa & proton versions.. also distro change just to ensure.. it still crashes my entire system on 5700xt and Ryzen 3500x

**Comment by SniperJoe** [comment]
> > Replying to [#8107 (comment)](https://github.com/ValveSoftware/Proton/issues/8107#issuecomment-2632359448)
> 
> So you're saying it gets the runtime working, but the overlay itself doesn't work properly?
> 
> Also, does this workaround work in other games like GoT?

Honestly, I'm not sure about overlay. I've been testing the game only after Sony cancelled the demand of having a PSN account. 

GoT should work with this workaround if you had the same issue (PsPcSdkRuntimeInstaller.msi halting upon installation) before, but I'm certain I've read somewhere that PSSDK is not distributed with GoT and all needed functions are built into the game itself instead.

**Comment by SniperJoe** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/8107#issuecomment-2725915147

My issue turned out to be Nvidia drivers issue, and GoW was not the only game suffering from it. All games on my machine (even including Linux native) were stuttering. I resolved it by purchasing an AMD card 😂 So I guess it's not your case. If I were you, I would try to set up Proton to log into some file so you can read it even after system crash.

**Comment by kisak-valve** [comment]
God of War Ragnarok core pinning issue (seems to be on the Proton side)

Issue transferred from https://github.com/ValveSoftware/Proton/issues/8566.
@camberkenpas posted on 2025-03-27T03:32:00:

# Compatibility Report
- Name of the game with compatibility issues: God of War Ragnarök
- Steam AppID of the game: 2322010

## System Information
- GPU: RTX 4090
- Video driver version: 570.124.04
- Kernel version: 6.13.8
- Link to full system information report as [Gist](https://gist.github.com/camberkenpas/ee4fdd8a4dbefd5b60a10c3f69e8b07a):
- Proton version: 9.0-4, Hotfix & Experimental (both as of 2025-03-26)

## I confirm:
- [x ] that I haven't found an existing compatibility report for this game.
- [x ] that I have checked whether there are updates for my system available.

[steam-2322010.log.gz](https://github.com/user-attachments/files/19478955/steam-2322010.log.gz)

## Symptoms 
When running the game through gamemoderun using core pinning, the game will crash shorty after launching. Usually pretty early on, but sometimes a bit longer.
In ~/.config/gamemode.ini if I set  pin_cores=0-15 under the cpu section, this happens. If I use core parking instead, there's no problem. Other games don't have this issue so it seems specific to GoW: Ragnarok. I've tested with other Sony games such as Days Gone, Ghost of Tsushima, Spider-Man: Miles Morales without issue. Also all non-Sony games I've tested are unaffected.

As it doesn't happen other games, this is likely a bug that gamemode is simply able to trigger through pinning. Maybe it's a DRM check of some kind in GoW?

## Reproduction
1. Edit `~/.config/gamemode.ini`
2.  Set `pin_cores=0-15` under the CPU section (unsure if this affects single CCD systems, etc).
3. Run God of War: Ragnarok with the following launch options: `SteamDeck=1 gamemoderun %command%`

For now I can keep using core parking, but it would be nice to have the rest of my cores available for other tasks as I play.

**Comment by kisak-valve** [comment]
Hello @camberkenpas, give `WINE_CPU_TOPOLOGY` like it's discussed in #5927 a try as an alternative.

**Comment by camberkenpas** [comment]
@kisak-valve Setting `WINE_CPU_TOPOLOGY` doesn't fix the issue, regardless of the value. I've tried it even with all 32 threads.

I also found that if I don't put quotes around the value, the game won't start. It appears that anything after the semi is considered separate (which does make sense). 

Finally, I tried removing all parking and pinning, the game still crashes. It's stable with parking, but that's it. Which means the pinning has been a red herring.

When playing this game on a 7950x in the past, I didn't use any sort of pinning/parking and it ran fine... So this could be an issue with the 9950x3d and 9950x (I've tested both). I still have the system with the 7950x so I'll see if that still works without any parking/pinning (maybe it broke with a certain update to Proton?)... but now that box has a Radeon 9070 XT so it's possible we could still see crashes there due to less mature drivers.

**Comment by camberkenpas** [comment]
This issue does NOT occur on 9950x CPU as there is no automatic core pinning from gamemode for the non-x3d CPUs. 

However, leaving gamemode with the default core and pin settings, gamemode automatically pins to threads 0-7,16-23. This still results in GoW crashing. However, if I _park_ to the same 0-7,16-23 threads... the game also still crashes.

If I set `pin_cores=0-31` in the gamemode config (effectively disabling any automatic core parking/pinning) the game is stable on my 9950x3d system like it is on the 9950x and 7950x systems on default settings (which gamemode applies no core parking/pinning by default).

It seems the issue is related to parking/pinning with non-sequential threads which is unfortunate when it seems at least on some CPU's you need to use non-sequential threads for optimal performance!

If it's of any value, I can test parking with different sets of threads and as well as different numbers of threads, etc. Just let me know.

I'm in a unique position to test across 3 different systems in my house (all with Linux available):
- 9950x3d + RTX 4090 (my personal system and the easiest/quickest for me to test again)
- 9950x + RTX 4090
- 7950 + Radeon RX 9070 XT

Edit: Big corrections around core pinning/parking! gamemode auto detects pinning to the v-cache cores 9950x3d. While the comments in the default .ini default call out the 7950x3d specifically... I've verified in the code that this isn't hard coded and detection for mixed cache CPU's is done dynamically.This also means at least on the 9950x3d, Linux sees the threads of the first 8 cores 0-7,16-23 rather than at 0-15! I've updated the above to reflect my findings.

**Comment by pannal** [comment]
Confirmed on a 9950X3D. As long as you stay within consecutive threads it seems to work, at least with `WINE_CPU_TOPOLOGY=8` (the first 8 non-SMT cores).

**Comment by telepati** [comment]
The game works without any problems, but Dualsense detects it as an XBOX Controller Input. Is there any way to fix this?

**Comment by SniperJoe** [comment]
@telepati Perform actions from my comment: https://github.com/ValveSoftware/Proton/issues/8107#issuecomment-2642168026
After this the game will be able to use libScePad for DS instead of XInput

**Comment by BjarneSeger** [comment]
> For those who are interested in running the game with Sony gamepad, after 4 days of debugging & disassembling I came to a solution:
> 
>     1. Open regedit for the prefix you run the game on
> 
>     2. Go to HKEY_LOCAL_MACHINE\System\CurrentControlSet\Enum\HID
> 
>     3. In folder under name like "VID_054C&PID_{your_device_pid}", open the first child folder (it's name must be like "256&00:00:00:00:00:00&0&0&0")
> 
>     4. Create a string parameter with name "ContainerId" and value "{e27ebefb-ddd4-11ef-abea-40b076a40d44}" in the folder

@SniperJoe Does this still work? It seems like without using Steam Input, DualSense controllers are just not detected at all by GoWR, while The Last of Us or Space Marine 2 work fine. Tried different Proton versions with your fix but none of them even got the controller to work in that game at all

**Comment by Saladin1812** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/8107#issuecomment-2763047637

I have 3600x + 5700 XT.. Could this be related I tried everything.. This is the only game that is crashing and taking my whole system with it. And it happens few seconds into the main menu and sometimes just few seconds after I am in game. I play other highly demanding games no issues. This one just crashes. Any logs that I can get ? I am unable to get proton logs themselves as the whole system crashes. There is a sound crackling behind and then boom its gone.

**Comment by camberkenpas** [comment]
I'm running Ubuntu 25.04. The game doesn't detect my Dualsense or Xbox Core controllers. I'm able to use these controllers in other games (provided the games support them). My first time starting up, I got a message basically saying the controller was supported, but since the game was running with Steam Input, I may not see the correct button prompts. I've tried disabling Steam Input, forcing it, etc. Nothing works.

Another very annoying problem, which I suspect is related, is that I get accessibility narration where the game is constantly telling me what I'm doing. I haven't found a way to turn this off.

Performance seems acceptable with a few occasional hitches/stutters (some but not all seems potentially related to traversal). Just trying the first stage, I seem to get 100-130+ fps. I haven't tried in Windows yet so I have no idea how good this performance is (but would certainly more than playable if I could use a controller).

I sometimes see short delays of texture loads... Could be that it happens in Windows too because the game runs on several consoles which all have unified memory. That being said, the game is running off of a Gen 4 NVME and I'm using an RTX 4090.

**Comment by simifor** [comment]
@SaladinAyyub even if the whole system crashes, it should still create a log if the game works for a bit. If you launch the game with `PROTON_LOG=1 %command%` a file called `steam-2322010.log` in your home folder.
Also, after your system hangs and you're forced to reboot, you can get logs from the previous session by using the following commands in your terminal:
`journalctl -b -1 > journal.log` and `journalctl -b -1 -k > dmesg.log`

**Comment by runar-work** [comment]
I haven't been to reproduce any issues with RX 5700 XT and various distros on my system, at least (7900 with Ubuntu 24.04 LTS, Bazzite 42 and Arch Linux). I've tried a variety of graphics settings and Mesa versions, so I haven't found any clues there either.

**Comment by zeuscaptain77-dot** [comment]
> # Compatibility Report
> * Name of the game with compatibility issues: God of War Ragnarok
> * Steam AppID of the game: 2322010
> 
> ## System Information
> * GPU:  Nvidia 4090
> * Video driver version: 560.35.03
> * Kernel version: 6.10.10
> * Link to full system information report: [System info gist](https://gist.github.com/hjpaul7/ea6405becfc2643394a0302ac8758d39)
> * Proton version: `1721412568 experimental-9.0-20240719`
> 
> ## I confirm:
> * [x]  that I haven't found an existing compatibility report for this game.[x]  that I have checked whether there are updates for my system available.
> 
> [steam-2322010.log](https://github.com/user-attachments/files/17062139/steam-2322010.log)
> 
> ## Symptoms
> * Hit with "The program cannot start because the PlayStation PC SDK runtime is missing from you computer."
> 
> ## Reproduction
> * Launch game

**Comment by Saladin1812** [comment]
> I haven't been to reproduce any issues with RX 5700 XT and various distros on my system, at least (7900 with Ubuntu 24.04 LTS, Bazzite 42 and Arch Linux). I've tried a variety of graphics settings and Mesa versions, so I haven't found any clues there either.

I upgraded my hardware and its fixed.. so probably was a hardware issue on my side.

**Comment by chenxiaolong** [comment]
I was able to get this working well with the PS5 controller (especially haptics) with a few changes:

* Instead of using `SteamDeck=1`, I'm using https://github.com/LNDF/PsPcSdkEmulator to disable the PlayStation SDK. `SteamDeck=1` normally already disables the SDK, but unfortunately causes the game to always show the Steam Deck/XBox button icons instead of the PlayStation ones.
* For haptics, I built Proton with the patches from https://github.com/xzn/proton-ds5-haptic. [`0006-setupapi-SetupDiGetDeviceRegistryProperty-now-return.patch`](https://github.com/xzn/proton-ds5-haptic/blob/5e48b0d2c1522a063371703eec16ec41c6857ae2/0006-setupapi-SetupDiGetDeviceRegistryProperty-now-return.patch) seems to be the important one, though I didn't try building with just that single patch.
* For folks on Linux distros with a newer alsa-ucm-conf package, like Fedora or Arch, https://github.com/ValveSoftware/Proton/issues/5900#issuecomment-3696038585 is also necessary to fix DS5 haptics (in any game).

**Comment by DanielYang59** [comment]
## Don't fully work with 3840 x 2560 (3:2) 4K+ Huawei Mateview 28 monitor

- https://github.com/ValveSoftware/Proton/issues/9729

## System Information
- GPU:  AMD Radeon RX 6800
- Video driver version: Mesa 25.2.8-0ubuntu0.24.04.1
- Kernel version: 6.17.0-14-generic
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/DanielYang59/7e6ccffb8986af7cebf3126c75b921e0
- Proton version: 10.0-4

> [!NOTE]
> I could help testing other proton versions, let me know if this would be helpful.

## Symptoms 

I have a [Huawei Mateview 28](https://consumer.huawei.com/ch/mateview/) which has a weird 3:2 ratio at 3840 * 2560 @ 60Hz (via DP cable). I noticed some games don't fully work in "full screen", more details below.


- God of War Ragnarök (2322010): <img width="341" height="117" alt="Image" src="https://github.com/user-attachments/assets/42c07558-4d90-4f65-a0f1-2e0d7d900277" />
    - "hacked" following comment via `SteamDeck=1 %command%` in https://github.com/ValveSoftware/Proton/issues/8107
    - full screen mode capped at "3840 x 2400"
    - windowed mode capped at "3700 x 2462" (maybe normal considering menu bar), with boarder to the top/bottom
 

<img width="4096" height="3072" alt="Image" src="https://github.com/user-attachments/assets/a11bd359-abdc-4cc3-a61e-1f4d03324528" />
<img width="4096" height="3072" alt="Image" src="https://github.com/user-attachments/assets/5167d398-cf7e-443f-b7d8-1d346a1cc18f" />
<img width="4096" height="3072" alt="Image" src="https://github.com/user-attachments/assets/61b2fec2-183b-430d-bd7a-36ddb7e8e695" />

**Comment by Saladin1812** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/8107#issuecomment-4359224078

Try enabling wayland -> `PROTON_ENABLE_WAYLAND=1 %command%` use this flag in launch options. On some DEs & Window managers this works better. On some without it. @DanielYang59 and then try the ingame setting, that should fix it.

**Comment by DanielYang59** [comment]
Hi @Saladin1812 thanks for the comment, I just tried `PROTON_ENABLE_WAYLAND=1 %command%` alone wasn't enough to get the game running (got the PSPC error), with `SteamDeck=1 PROTON_ENABLE_WAYLAND=1 %command%` I seems to get the same behaviour (resolution in full screen capped at "3840 x 2400")
```
>>> echo $XDG_SESSION_TYPE
wayland
```

**Comment by arthuralvesdev** [comment]
Just a heads up for anyone playing with high-polling-rate mice (1000 Hz): moving the mouse can cause severe drops to 10 FPS due to how the game calls GetRawInputData with a null size buffer, triggering constant exceptions in wine's win32u. Lowering the mouse polling rate to 500 Hz or enabling Steam Input fixes the framerate drops.
